qt/scientific_interfaces/Indirect/test/IndirectPlotOptionsModelTest.h

```diff
@@ -0,0 +1,300 @@
+//
+// Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#ifndef MANTIDQT_INDIRECTPLOTOPTIONSMODELTEST_H_
+#define MANTIDQT_INDIRECTPLOTOPTIONSMODELTEST_H_
+
+#include <cxxtest/TestSuite.h>
+#include <gmock/gmock.h>
+
+#include "IndirectPlotOptionsModel.h"
+
+#include "MantidAPI/AnalysisDataService.h"
+#include "MantidAPI/ITableWorkspace.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidDataObjects/Workspace2D.h"
+#include "MantidKernel/WarningSuppressions.h"
+#include "MantidTestHelpers/WorkspaceCreationHelper.h"
+
+using namespace Mantid::API;
+using namespace Mantid::DataObjects;
+using namespace MantidQt::CustomInterfaces;
+using namespace testing;
+
+namespace {
+
+std::string const WORKSPACE_NAME = "WorkspaceName";
+std::string const WORKSPACE_INDICES = "0-2,4";
+
+MatrixWorkspace_sptr convertWorkspace2DToMatrix(Workspace2D_sptr workspace) {
+  return boost::dynamic_pointer_cast<MatrixWorkspace>(workspace);
+}
+
+MatrixWorkspace_sptr
+createMatrixWorkspace(std::size_t const &numberOfHistograms,
+                      std::size_t const &numberOfBins) {
+  return convertWorkspace2DToMatrix(WorkspaceCreationHelper::create2DWorkspace(
+      numberOfHistograms, numberOfBins));
+}
+
+TableWorkspace_sptr createTableWorkspace(std::size_t const &size) {
+  return boost::make_shared<TableWorkspace>(size);
+}
+
+/// Mock object to mock an IndirectTab
+class ParentTab : public IndirectTab {
+private:
+  void setup() override{};
+  void run() override{};
+  bool validate() override { return true; };
+};
+
+} // namespace
+
+GNU_DIAG_OFF_SUGGEST_OVERRIDE
+
+/// Mock object to mock an IndirectTab
+class MockIndirectPlotter : public IndirectPlotter {
+public:
+  /// Public Methods
+  MOCK_METHOD2(plotSpectra, void(std::string const &workspaceName,
+                                 std::string const &workspaceIndices));
+  MOCK_METHOD2(plotBins, void(std::string const &workspaceName,
+                              std::string const &binIndices));
+  MOCK_METHOD1(plotContour, void(std::string const &workspaceName));
+  MOCK_METHOD2(plotTiled, void(std::string const &workspaceName,
+                               std::string const &workspaceIndices));
+};
+
+GNU_DIAG_ON_SUGGEST_OVERRIDE
+
+class IndirectPlotOptionsModelTest : public CxxTest::TestSuite {
+public:
+  IndirectPlotOptionsModelTest() : m_ads(AnalysisDataService::Instance()) {
+    m_ads.clear();
+  }
+
+  static IndirectPlotOptionsModelTest *createSuite() {
+    return new IndirectPlotOptionsModelTest();
+  }
+
+  static void destroySuite(IndirectPlotOptionsModelTest *suite) {
+    delete suite;
+  }
+
+  void setUp() override {
+    m_plotter = std::make_unique<NiceMock<MockIndirectPlotter>>();
+    m_model =
+        std::make_unique<IndirectPlotOptionsModel>(std::move(m_plotter.get()));
```

Comment:
I'm not sure the move is necessary here and I think it ends up being owned both locally and by the model? Would it be better to just create it as a raw pointer here if the model takes ownership?